db/osqlshadtbl.c

```diff
@@ -1399,13 +1436,15 @@ int osql_save_qblobs(struct BtCursor *pCur, struct sql_thread *thd,
 
         if (blobs[i].exists == 1) {
 
+            (void)odhfy_blob_buffer(pCur->db, blobs + i, i);
```

Comment:
I think this is fine in general. Saving compressed blobs in temp table favors osql resubmission but not transactions (read committed or higher) who read their own inserts. I am not saying this is a bottleneck but it might worth benchmarking.

Reply:
Gotcha. Will do a benchmark and post results in the pull request.